test/SecTester.Repeater.Tests/Runners/HttpRequestRunnerTests.cs

```diff
@@ -0,0 +1,336 @@
+      .Respond(HttpStatusCode.OK, JsonContentType, JsonContent);
+
+    // act
+    var result = await sut.Run(request);
+
+    // assert
+    _mockHttp.VerifyNoOutstandingExpectation();
+    result.Should().BeEquivalentTo(new
+    {
+      StatusCode = (int)HttpStatusCode.OK,
+      Body = JsonContent
+    });
+  }
+
+  [Fact]
+  public async Task Run_ReturnsResultWithDecodedBody()
+  {
+    // arrange
+    var sut = CreateSut();
+    var encoding = Encoding.GetEncoding("utf-16");
+    var expectedByteLength = Buffer.ByteLength(encoding.GetBytes(HtmlBody));
+    var request = new RequestExecutingEvent(new Uri(Url));
+    var content = new StringContent(HtmlBody, encoding, HtmlContentType);
+    _mockHttp.Expect(Url).Respond(HttpStatusCode.OK, content);
+
+    // act
+    var result = await sut.Run(request);
+
+    // assert
+    _mockHttp.VerifyNoOutstandingExpectation();
+    result.Should().BeEquivalentTo(new
+    {
+      Headers = new[] { new KeyValuePair<string, string[]>(ContentTypeFieldName, new[] { HtmlContentTypeWithCharSet }), new KeyValuePair<string, string[]>(ContentLengthFieldName, new[] { $"{expectedByteLength}" }) },
+      Body = HtmlBody
+    }, options => options.ExcludingMissingMembers().IncludingNestedObjects());
+  }
+
+  [Fact]
+  public async Task Run_ReturnsResultWithError_WhenRequestTimesOut()
+  {
+    // arrange
+    var sut = CreateSut(new RequestRunnerOptions
+    {
+      Timeout = TimeSpan.Zero
+    });
+    var request = new RequestExecutingEvent(new Uri(Url));
+    _mockHttp.Expect(Url)
+      .Respond(async () =>
+      {
+        await Task.Delay(5);
+
+        return new HttpResponseMessage(HttpStatusCode.OK);
+      });
+
+    // act
+    var result = await sut.Run(request);
+
+    // assert
+    _mockHttp.VerifyNoOutstandingExpectation();
+    result.Should().BeEquivalentTo(new
+    {
+      Message = "The operation was canceled."
+    });
+  }
+
+  [Fact]
+  public async Task Run_MaxContentLengthIsLessThan0_SkipsTruncating()
+  {
+    // arrange
+    var sut = CreateSut(new RequestRunnerOptions { MaxContentLength = -1 });
+    var request = new RequestExecutingEvent(new Uri(Url));
+    var body = string.Concat(Enumerable.Repeat("x", 5));
+    _mockHttp.Expect(Url).Respond(HttpStatusCode.OK, CustomContentType, body);
+
+    // act
+    var result = await sut.Run(request);
+
+    // assert
+    result.Should().BeEquivalentTo(new
+    {
+      Body = body
+    });
+  }
+
+  [Fact]
+  public async Task Run_NoContentStatusReceived_SkipsTruncating()
+  {
+    // arrange
+    var sut = CreateSut();
+    var request = new RequestExecutingEvent(new Uri(Url));
+    _mockHttp.Expect(Url).Respond(HttpStatusCode.NoContent);
+
+    // act
+    var result = await sut.Run(request);
+
+    // assert
+    result.Should().BeEquivalentTo(new
+    {
+      StatusCode = (int)HttpStatusCode.NoContent
```

Comment:
```suggestion
      StatusCode = (int)HttpStatusCode.NoContent,
      Body = ""
```
It's not clear the result of the operation for empty content, BTW why it is not null?

Reply:
> BTW why it is not null?
Maybe because this is the contract between the engine and any other client?